src/communication/index.ts

```diff
@@ -1,108 +1,69 @@
-import {
-  InterfaceMessageIds,
-  InterfaceMessageEvent,
-  SentSDKMessage,
-  SDKMessageIds,
-  SDKMessageToPayload,
-  RequestId,
-  InterfaceMessageToPayload,
-  Communicator,
-} from '../types';
-import { INTERFACE_MESSAGES } from './messageIds';
+import semver from 'semver';
+import { InterfaceMessageEvent, Communicator, Methods, Response } from '../types';
+import { MessageFormatter } from './messageFormatter';
 
-class InterfaceCommunicator implements Communicator {
-  private allowedOrigins: RegExp[] = [];
+// eslint-disable-next-line
+type Callback = (response: any) => void;
 
-  constructor(allowedOrigins: RegExp[]) {
+class PostMessageCommunicator implements Communicator {
+  private readonly allowedOrigins: RegExp[] | null = null;
+  private callbacks = new Map<string, Callback>();
+
+  constructor(allowedOrigins: RegExp[] | null = null) {
     this.allowedOrigins = allowedOrigins;
 
     window.addEventListener('message', this.onParentMessage);
   }
 
-  private isValidMessage({ origin, data }: InterfaceMessageEvent): boolean {
-    const emptyOrMalformed = !data || !data.messageId;
-    const unknownOrigin = this.allowedOrigins?.find((regExp) => regExp.test(origin)) === undefined;
-    const sameOrigin = origin === window.origin;
+  private isValidMessage = ({ origin, data, source }: InterfaceMessageEvent): boolean => {
+    const emptyOrMalformed = !data;
+    const sentFromParentEl = source === window.parent;
+    const allowedSDKVersion = typeof data.version !== 'undefined' ? semver.gte(data.version, '1.0.0') : false;
```

Comment:
does it make sense to handle versions? 
If there is a breaking change in some method, the safe-app updating to the new sdk version will need to refactor the code. They can always continue using an old version if they don't want to refactor.

Reply:
nevermind, I see why you did this.